### ISSUE TITLE
Replicate the product on Javascript

### ISSUE THREAD

**Issue by RichardAwoyemi** [OpenRarity/open-rarity#53]
It's been amazing to use so far on Python. That said, is it possible to recreate it on JS too?

It would also be great if we could still use NPM / Yarn to handle the packages similarly to pip. 

Thanks

**Comment by impreso** [comment]
`said, is it possible to recreate it on JS too?` ->  Great idea @RichardAwoyemi. On JS native support , we have something in work that will help you - stay tuned! I will update the ticket once it's available. 

`it would also be great if we could still use NPM / Yarn to handle the packages similarly to pip.`  -> you mean add support like it's described here https://classic.yarnpkg.com/en/package/python.node?

**Comment by samstr** [comment]
A TypeScript release would be amazing. I intend to use this on Rareboy.com

**Comment by emorling** [comment]
subscribe

> `is it possible to recreate it on JS too?` -> Great idea @RichardAwoyemi. On JS native support , we have something in work that will help you - stay tuned! I will update the ticket once it's available.
> 
> `it would also be great if we could still use NPM / Yarn to handle the packages similarly to pip.` -> you mean add support like it's described here https://classic.yarnpkg.com/en/package/python.node?

Awesome any ETA on JS version?

**Comment by emorling** [comment]
> While we are researching Node/JS support, will cloud endpoint API suffice your needs @emorling @samstr @RichardAwoyemi ?

Yes, but it depends on the rate limit. Will we be sending in all the metadata for a single collection in one request?

**Comment by impreso** [comment]
@emorling no metadata needed for this particular endpoint

**Comment by samstr** [comment]
I had assumed the logic would be client side, rather than an API endpoint. 
It would be nice to prepare metadata as a local object/json and simply run it through a function to generate scores. Some projects I plan to generate with haven't minted yet, and I assume the endpoint would just take a `contractAddress` and give the openrarity from an already minted collection?

**Comment by impreso** [comment]
@samstr great idea! 

API endpoint was developed for simplicity of the data access and is complementary to the open-source library.

For not minted contracts you could just pipe collection data from metadata into OpenRarity models, you don't need to read from blockchain to construct the data. Check out these developer docs on how we do it in Python: https://openrarity.gitbook.io/developers/quick-guides/integrating-openrarity-in-your-application. 

If you have the bandwidth to build this tool, we will be glad to figure out how to land the PR.

**Comment by jeevanpillay** [comment]
i'd be happy to jump in and contribute to the js version as well if needed. 

and yes @impreso, an API endpoint would be a good start, and then eventually moving into standalone npm would be a great.

**Comment by impreso** [comment]
Hi all, that was a fun week for the OpenRarity team here, thanks for all the support and love from the community. We are carefully listening for all feedback that ya'll give and actively working on our next release.

Here is promised update with the OpenSea API that provides rarity data right now on the [Assets API endpoint](https://docs.opensea.io/reference/asset-object#rarity-data)  . We are still figuring out how to support other languages than python, but this should give you a straight integration point for JS. 

One idea we had is to ship an OpenRarity docker image with an exposed REST endpoint to support all languages out there. Thoughts on this technical direction?

**Comment by emorling** [comment]
Hey @impreso, I’m impressed by your progress and devotion to this topic. I think a Docker Image would be great, and Iook forward to exploring the endpoints

**Comment by RichardAwoyemi** [comment]
> Hi all, that was a fun week for the OpenRarity team here, thanks for all the support and love from the community. We are carefully listening for all feedback that ya'll give and actively working on our next release.
> 
> Here is promised update with the OpenSea API that provides rarity data right now on the [Assets API endpoint](https://docs.opensea.io/reference/asset-object#rarity-data) . We are still figuring out how to support other languages than python, but this should give you a straight integration point for TS.
> 
> One idea we had is to ship an OpenRarity docker image with an exposed REST endpoint to support all languages out there. Thoughts on this technical direction?

Sorry I've been slow to respond in this thread but your efforts have been amazing @impreso! Wouldn't we need an OS API key to get the current Asset modal via the current endpoints?

The exposed docker endpoint would be a massive help. That way we could interact with it directly via js, or any other language.

**Comment by gregorym** [comment]
If you offered a CLI rather than an SDK for each language that would be the best IMO.

**Comment by emorling** [comment]
> If you offered a CLI rather than an SDK for each language that would be the best IMO.

That would not suffice for my use cases, when providing Rarity in a web site or chrome extension.

**Comment by gregorym** [comment]
> > If you offered a CLI rather than an SDK for each language that would be the best IMO.
> 
> That would not suffice for my use cases, when providing Rarity in a web site or chrome extension.

Ok but that's not everyone's use case. If you have a website, you can run the cli server side and expose it on your website. Same with a chrome extension.

**Comment by kongnet** [comment]
https://github.com/kongnet/openrarity

openRerity core algorithm JS implementation

I will add 100k tokens benchmark test.

**Comment by HIT2022** [comment]
Is this still on the roadmap? Would love to implement in a web app

**Comment by impreso** [comment]
Yes , absolutely. The team is heads down implementing the library. @dadashi to comment on this

**Comment by dadashi** [comment]
@HIT2022 yes we're actively working on this. Can you speak more to how you plan on using it in your web app? Are you planning to use it in the browser or backend?

**Comment by danielmedvec** [comment]
Any updates on the JS version?

**Comment by vikiival** [comment]
👀 gm gm
